packages/component-library/src/components/icons-media/mt-icon/mt-icon.vue

```diff
@@ -57,8 +59,10 @@ const styles = computed(() => {
 watch(
   () => props.name,
   (newName) => {
-    const [variant] = newName.split("-");
-    const iconName = newName.split("-").slice(1).join("-");
+    const variant = /^(solid|regular)/.test(newName) ? newName.split("-")[0] : props.mode;
```

Comment:
Should we be using regex in this component? I know it's a quite simple one, but imagine having dozens of icons being rendered at once times 3 RegEx per icon, could be interesting performance-wise.

Reply:
Or just use `split('-.)`: If you have two items, it's safe to assume that both `variant` and `name` are defined

Reply:
Ok, I added a computed property and replaced the regexes with `.startsWith()`